src/backend/distributed/deparser/deparse_function_stmts.c

```diff
@@ -0,0 +1,596 @@
+#else
+	Assert(stmt->objectType == OBJECT_FUNCTION || stmt->objectType == OBJECT_PROCEDURE);
+#endif
+
+	AppendAlterFunctionOwnerStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+/*
+ * AppendAlterFunctionOwnerStmt appends a string representing the AlterOwnerStmt to a buffer
+ */
+static void
+AppendAlterFunctionOwnerStmt(StringInfo buf, AlterOwnerStmt *stmt)
+{
+	ObjectWithArgs *func = castNode(ObjectWithArgs, stmt->object);
+
+#if (PG_VERSION_NUM < 110000)
+	appendStringInfo(buf, "ALTER FUNCTION ");
+#else
+	if (stmt->objectType == OBJECT_FUNCTION)
+	{
+		appendStringInfoString(buf, "ALTER FUNCTION ");
+	}
+	else
+	{
+		appendStringInfoString(buf, "ALTER PROCEDURE ");
+	}
+#endif
+
+	AppendFunctionName(buf, func);
+	appendStringInfo(buf, " OWNER TO %s;", RoleSpecString(stmt->newowner));
+}
+
+
+/*
+ * DeparseAlterFunctionDependsStmt builds and returns a string representing the AlterObjectDependsStmt
+ */
+const char *
+DeparseAlterFunctionDependsStmt(AlterObjectDependsStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->objectType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->objectType == OBJECT_FUNCTION || stmt->objectType == OBJECT_PROCEDURE);
+#endif
+
+	AppendAlterFunctionDependsStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+/*
+ * AppendAlterFunctionDependsStmt appends a string representing the AlterObjectDependsStmt to a buffer
+ */
+static void
+AppendAlterFunctionDependsStmt(StringInfo buf, AlterObjectDependsStmt *stmt)
+{
+	ObjectWithArgs *func = castNode(ObjectWithArgs, stmt->object);
+
+#if (PG_VERSION_NUM < 110000)
+	appendStringInfo(buf, "ALTER FUNCTION ");
+#else
+	if (stmt->objectType == OBJECT_FUNCTION)
+	{
+		appendStringInfoString(buf, "ALTER FUNCTION ");
+	}
+	else
+	{
+		appendStringInfoString(buf, "ALTER PROCEDURE ");
+	}
+#endif
+
+	AppendFunctionName(buf, func);
+	appendStringInfo(buf, " DEPENDS ON EXTENSION %s;", strVal(stmt->extname));
+}
+
+
+/*
+ * DeparseDropFunctionStmt builds and returns a string representing the DropStmt
+ */
+const char *
+DeparseDropFunctionStmt(DropStmt *stmt)
+{
+	StringInfoData str = { 0 };
+	initStringInfo(&str);
+
+#if (PG_VERSION_NUM < 110000)
+	Assert(stmt->removeType == OBJECT_FUNCTION);
+#else
+	Assert(stmt->removeType == OBJECT_FUNCTION || stmt->removeType == OBJECT_PROCEDURE);
+#endif
+
+	AppendDropFunctionStmt(&str, stmt);
+
+	return str.data;
+}
+
+
+/*
+ * AppendDropFunctionStmt appends a string representing the DropStmt to a buffer
+ */
+static void
+AppendDropFunctionStmt(StringInfo buf, DropStmt *stmt)
+{
+#if (PG_VERSION_NUM < 110000)
+	appendStringInfo(buf, "DROP FUNCTION ");
+#else
+	if (stmt->removeType == OBJECT_FUNCTION)
+	{
+		appendStringInfoString(buf, "DROP FUNCTION ");
+	}
+	else
+	{
+		appendStringInfoString(buf, "DROP PROCEDURE ");
+	}
+#endif
+
+	if (stmt->missing_ok)
+	{
+		appendStringInfoString(buf, "IF EXISTS ");
+	}
+
+	AppendFunctionNameList(buf, stmt->objects);
```

Comment:
Can you add a test which drops if exists multiple functions?`DROP FUNCTION IF EXISTS add(int,int), add(int);`